test/channel.js

```diff
@@ -225,6 +225,39 @@ test('event channel', assert => {
   assert.end()
 });
 
+test('unsubscribe event channel', assert => {
+  let unsubscribed = false;
+  let chan = eventChannel(() => () => {
+    unsubscribed = true;
+  });
+  chan.close();
+  assert.ok(unsubscribed, 'eventChannel should call unsubscribe when channel is closed')
+
+  unsubscribed = false;
+  chan = eventChannel((emitter) => {
+    emitter(END);
+    return () => {
+      unsubscribed = true;
+    }
+  });
+  assert.ok(unsubscribed, 'eventChannel should call unsubscribe when END event is emitted synchronously');
+
+  unsubscribed = false;
+  chan = eventChannel((emitter) => {
+    setTimeout(() => emitter(END), 0);
+    return () => {
+      unsubscribed = true;
+    }
+  });
+  chan.take(input => {
+    // console.log(input, unsubscribed);
+    assert.equal(input, END, 'should emit END event');
+    // assert.ok(unsubscribed, 'eventChannel should call unsubscribe when END event is emitted asynchronously');
```

Comment:
Fixed it on your branch, take a look at [this commit](https://github.com/baldwmic/redux-saga/commit/5a9108066c36728090c23364acc14a7e7b9f8137). 

I'll review the whole thing once more later and gonna merge it in. Thanks for the great work and noticing this.